gvc.go

```diff
@@ -133,74 +185,52 @@ func cleanup(path string, opts options) error {
 		if lastVendorPath == "" {
 			lastVendorPath = localPath
 		}
+		lastVendorPathDir := filepath.Dir(lastVendorPath)
 
 		keep := false
+
 		for _, name := range pkgList {
-			// If the file's parent directory is a needed package, keep it.
-			if !info.IsDir() && filepath.Dir(lastVendorPath) == name {
-				if opts.onlyCode {
-					if opts.noTests && strings.HasSuffix(path, "_test.go") {
-						keep = false
-						continue
-					}
-					for _, suffix := range codeSuffixes {
-						if strings.HasSuffix(path, suffix) {
-							keep = true
-						}
-					}
-					// Match keep patterns
-					for _, keepPattern := range opts.keepPatterns {
-						ok, err := doublestar.Match(keepPattern, lastVendorPath)
-						// TODO(sgotti) if a bad pattern is encountered stop here. Actually there's no function to verify a pattern before using it, perhaps just a fake match at the start will work.
-						if err != nil {
-							return fmt.Errorf("bad pattern: %q", keepPattern)
-						}
-						if ok {
-							keep = true
-						}
-					}
-				} else {
-					keep = true
-				}
+			// If a directory is a needed package then keep it
+			keep = keep || info.IsDir() && name == lastVendorPath
+
```

Comment:
just noticed than in the previous iteration you called break after setting keep to true. This avoided some unneeded looping on the other packages since the file was already marked to be kept. Did you removed this because it's not a big deal and to keep the code cleaner?